core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -94,26 +98,35 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
 
   def entries: Int = _entries
 
+  def length: Long = _length
+
   /**
    * Reset the size of the memory map and the underneath file. This is used in two kinds of cases: (1) in
    * trimToValidSize() which is called at closing the segment or new segment being rolled; (2) at
    * loading segments from disk or truncating back to an old segment where a new log segment became active;
    * we want to reset the index size to maximum index size to avoid rolling new segment.
    */
-  def resize(newSize: Int) {
+  def resize(newSize: Int): Boolean = {
```

Comment:
We should document the result type.

Reply:
Sure. I have added the Java doc as suggested.